### PR TITLE
Improve logging for guest agent issues

### PR DESCRIPTION
**Pull request by benank**
Fixes:
 - Support codes getting truncated in status blobs
 - `__log_telemetry_info` in ActionHandler not working due to code execution order
 - Changes EventsFolderExists in `__log_telemetry_info` to EventsFolderExistedPreviously to capture its initial state before the events folder may have been created

Adds:
 - Extra logging of the value of the guest agent supported environment variable
 - Environment variable code to give an idea of what failed when getting the guest agent env var to the extension (already existed in core)

### CODE REVIEW

File: src/extension/src/ActionHandler.py [hunk 1]
```diff
@@ -117,30 +117,33 @@ def setup_telemetry(self):
...
-        events_folder_exists = os.path.exists(events_folder) if events_folder is not None else False
-        env_var_supports_telemetry = self.telemetry_writer.is_agent_compatible()
-        telemetry_info = "[EventsFolder=\'{0}\'][EventsFolderExists={1}][EnvVar={2}]".format(
-            events_folder_str, str(events_folder_exists), env_var_supports_telemetry)
+        agent_env_var_code = self.telemetry_writer.agent_env_var_code
+        telemetry_info = "[EventsFolder={0}][EventsFolderExistedPreviously={1}][EnvVarCode={2}]".format(
+            events_folder_str, str(events_folder_previously_existed), str(agent_env_var_code))
 
-        if env_var_supports_telemetry is True:
+        if agent_env_var_code == 0:
```
Comment (kjohn-msft) [inline src/extension/src/ActionHandler.py:146]: This is a prime candidate for an enumeration defined in Constants. The meaning of 0, 1, 2 don't mean anything to someone trying to parse the code quickly, and encountering this in other files.

File: src/extension/src/TelemetryWriter.py [hunk 2]
```diff
@@ -65,17 +66,23 @@ def __ensure_message_restriction_compliance(self, full_message):
...
+            self.agent_env_var_code = 1
             return False
 
         features_keyvalue_list = json.loads(features_keyvalue_list_str)
-        return any(kv_pair for kv_pair in features_keyvalue_list if kv_pair['Key'] == Constants.TELEMETRY_EXTENSION_PIPELINE_SUPPORTED_KEY)
+        telemetry_supported_key_exists = any(kv_pair for kv_pair in features_keyvalue_list if kv_pair['Key'] == Constants.TELEMETRY_EXTENSION_PIPELINE_SUPPORTED_KEY)
+        if telemetry_supported_key_exists is False:
+            self.logger.log_error('Guest agent does not support telemetry. [Error=Key not found: {0}]'.format(Constants.TELEMETRY_EXTENSION_PIPELINE_SUPPORTED_KEY))
+            self.logger.log_warning('Env var value: \n{0}'.format(json.dumps(features_keyvalue_list, indent=4, sort_keys=True)))
+            self.agent_env_var_code = 2
```
Comment (kjohn-msft) [inline src/extension/src/TelemetryWriter.py:83]: Same comment about Enum -- look at PatchOperationErrorCodes as an example for what was meant.

File: src/extension/src/TelemetryWriter.py [hunk 3]
```diff
@@ -65,17 +66,23 @@ def __ensure_message_restriction_compliance(self, full_message):
...
         if features_keyvalue_list_str is None:
+            self.agent_env_var_code = 1
             return False
 
         features_keyvalue_list = json.loads(features_keyvalue_list_str)
-        return any(kv_pair for kv_pair in features_keyvalue_list if kv_pair['Key'] == Constants.TELEMETRY_EXTENSION_PIPELINE_SUPPORTED_KEY)
+        telemetry_supported_key_exists = any(kv_pair for kv_pair in features_keyvalue_list if kv_pair['Key'] == Constants.TELEMETRY_EXTENSION_PIPELINE_SUPPORTED_KEY)
+        if telemetry_supported_key_exists is False:
+            self.logger.log_error('Guest agent does not support telemetry. [Error=Key not found: {0}]'.format(Constants.TELEMETRY_EXTENSION_PIPELINE_SUPPORTED_KEY))
+            self.logger.log_warning('Env var value: \n{0}'.format(json.dumps(features_keyvalue_list, indent=4, sort_keys=True)))
```
Comment (kjohn-msft) [inline src/extension/src/TelemetryWriter.py:82]: Do not write multiple lines when a single line will suffice as, if and when telemetry works, this contributes to throttling. Also log_debug - this is not a customer concern that needs to surface in the API. The customer facing concern is that the agent version did not meet the min bar. (If it did meet the min bar, it's an issue internal to us and the information flow needs to be adapted to that.)

File: src/extension/src/file_handlers/ExtOutputStatusHandler.py [hunk 4]
```diff
@@ -210,12 +213,14 @@ def add_error_to_status(self, message, error_code=Constants.PatchOperationErrorC
...
-    def __ensure_error_message_restriction_compliance(full_message):
+    def __ensure_error_message_restriction_compliance(self, full_message):
         """ Removes line breaks, tabs and restricts message to a character limit """
         message_size_limit = Constants.STATUS_ERROR_MSG_SIZE_LIMIT_IN_CHARACTERS
         formatted_message = re.sub(r"\s+", " ", str(full_message))
-        return formatted_message[:message_size_limit-3] + '...' if len(formatted_message) > message_size_limit else formatted_message
+        if full_message not in self.__ignore_error_message_restriction_compliance_strings:
+            return formatted_message[:message_size_limit - 3] + '...' if len(formatted_message) > message_size_limit else formatted_message
+        else:
+            return formatted_message
```
Comment (kjohn-msft) [inline src/extension/src/file_handlers/ExtOutputStatusHandler.py:223]: Ensure UT coverage.

File: src/core/src/service_interfaces/StatusHandler.py [hunk 5]
```diff
@@ -681,12 +684,14 @@ def add_error_to_status(self, message, error_code=Constants.PatchOperationErrorC
...
-    def __ensure_error_message_restriction_compliance(full_message):
+    def __ensure_error_message_restriction_compliance(self, full_message):
         """ Removes line breaks, tabs and restricts message to a character limit """
         message_size_limit = Constants.STATUS_ERROR_MSG_SIZE_LIMIT_IN_CHARACTERS
         formatted_message = re.sub(r"\s+", " ", str(full_message))
-        return formatted_message[:message_size_limit-3] + '...' if len(formatted_message) > message_size_limit else formatted_message
+        if full_message not in self.__ignore_error_message_restriction_compliance_strings:
+            return formatted_message[:message_size_limit - 3] + '...' if len(formatted_message) > message_size_limit else formatted_message
+        else:
+            return formatted_message
```
Comment (kjohn-msft) [inline src/core/src/service_interfaces/StatusHandler.py:694]: Ensure UT coverage.

**Review by kjohn-msft — CHANGES_REQUESTED** [review]
Comments inline.

File: src/extension/src/ActionHandler.py (hunk 1 above)
Comment (benank) [reply src/extension/src/ActionHandler.py:146]: Done.

File: src/extension/src/file_handlers/ExtOutputStatusHandler.py (hunk 4 above)
Comment (benank) [reply src/extension/src/file_handlers/ExtOutputStatusHandler.py]: Removed for extension - only necessary in core.

**Review by kjohn-msft — APPROVED** [review]

**Pull request merged**